toolchain/lower/file_context.cpp

```diff
@@ -184,6 +184,8 @@ auto FileContext::GetOrCreateFunction(SemIR::FunctionId function_id,
   // TODO: Add this function to a list of specific functions whose definitions
   // we need to emit.
   specific_functions_[specific_id.index] = result;
+  // TODO: Should this be a pair of <function_id, specific_id> ?
+  specific_function_definitions_.push_back(specific_id);
```

Comment:
What's the difference between `specific_functions_` and ``specific_function_definitions_`? (perhaps `specific_function_definitions_` contains only the `specific_functions_` entries that are non-zero? and maybe that would suffice - to walk `specific_functions_` visiting the non-zero entries?)

Reply:
These are specific functions for which a function declaration was emitted, but no function definition. 
I think there would be a way to check that using specific_function_, and that is by traversing all of specific_function_ and checking if an llvm::Function corresponds to it. The issue with this, is that it's possible to create new declarations (so consequently new definitions needed) while lowering a definition, so that list would need to be re-traversed. That's also why (in the dependent patch) when traversing specific_function_definitions_, it cannot use a range iterator.